packages/react/dropdown-menu/src/DropdownMenu.tsx

```diff
@@ -168,19 +168,22 @@ const DropdownMenuTrigger = React.forwardRef<DropdownMenuTriggerElement, Dropdow
           disabled={disabled}
           {...triggerProps}
           ref={composeRefs(forwardedRef, context.triggerRef)}
-          // Handle anything that the browser considers a click for the element type if
-          // not using pointer e.g. Space keyup and Enter keydown
-          onClick={composeEventHandlers(props.onClick, () => context.onOpenChange(true))}
           onPointerDown={composeEventHandlers(props.onPointerDown, (event) => {
             // only call handler if it's the left button (mousedown gets triggered by all mouse buttons)
             // but not when the control key is pressed (avoiding MacOS right click)
             if (!disabled && event.button === 0 && event.ctrlKey === false) {
               // prevent trigger focusing when opening
               // this allows the content to be given focus without competition
               if (!context.open) event.preventDefault();
-              context.onOpenToggle();
+              context.onOpenChange(true);
```

Comment:
This change allows us to remove all the code here https://github.com/radix-ui/primitives/pull/1082/files#diff-5e95ef32039f278b6120df2487c18e5ada5be130bb884c0d36d89c0f20ee41ddL291-L299

Instead this happens:

- Pointer down opens
- Pointer down to close fires open event again but;
- Pointer down bubbles to document which triggers `onInteractOutside`
- Outside interaction closes it

Sooo, pointer down on trigger always opens, outside interaction always closes regardless of modality.

Reply:
I really like this approach, though the mechanics of closing are a bit more implicit than we usually have with triggers. Do you think the code is clear enough as is? or would an inline comment explaining help us in the future?